.github/workflows/ci.yml

```diff
@@ -135,6 +145,98 @@ jobs:
         env:
           CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
 
+  e2e-tests:
+    runs-on: ubuntu-latest
+    needs: frontend-build
+    strategy:
+      fail-fast: false
+      matrix:
+        browser:
+          - chromium
+          - firefox
+          - webkit
+    
+    name: End-to-end tests, ${{ matrix.browser }}
+
+    defaults:
+      run:
+        working-directory: backend
+
+    services:
+      postgres:
+        image: postgis/postgis:14-3.4
+        env:
+          POSTGRES_HOST_AUTH_METHOD: trust
+        ports:
+          - 5432:5432
+        # Needed because the postgres container does not provide a healthcheck
+        options:
+          --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+          --name postgres
+      redis:
+        image: redis:6
+        ports:
+          - 6379:6379
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up backend environment
+        uses: maykinmedia/setup-django-backend@v1.1
+        with:
+          apt-packages: 'gettext postgresql-client libgdal-dev libyaml-dev gdal-bin'
+          python-version: '3.12'
+          optimize-postgres: 'yes'
+          pg-service: 'postgres'
+          setup-node: 'yes'
+          npm-ci-flags: '--legacy-peer-deps'
+          working-directory: backend
+          nvmrc-custom-dir: backend
+
+      # See https://playwright.dev/python/docs/ci#caching-browsers
+      - name: Cache Playwright browser
+        id: cache-browser
+        uses: actions/cache@v4
+        with:
+          path: /home/runner/.cache/ms-playwright
+          key:
+            ${{ runner.os }}-${{ matrix.browser }}-playwright-${{ hashFiles('requirements/ci.txt') }}
+
+      - name: Install playwright deps
+        run: playwright install --with-deps ${{ matrix.browser }}
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: build
+          path: ${{ github.workspace }}/frontend/build
+
+      - name: Run tests
+        working-directory: backend
+        run: |
+          ln -s ${{ github.workspace }}/frontend/build/static/css ${{ github.workspace }}/backend/src/openarchiefbeheer/static/css
+          ln -s ${{ github.workspace }}/frontend/build/static/js ${{ github.workspace }}/backend/src/openarchiefbeheer/static/js
+          ln -s ${{ github.workspace }}/frontend/build/index.html ${{ github.workspace }}/backend/src/openarchiefbeheer/templates/index.html
+          python src/manage.py compilemessages
+          python src/manage.py collectstatic --noinput --link
+          python src/manage.py test openarchiefbeheer --tag=e2e
+        env:
+          DJANGO_SETTINGS_MODULE: openarchiefbeheer.conf.ci
+          SECRET_KEY: dummy
+          DB_USER: postgres
+          DB_PASSWORD: ''
+          PLAYWRIGHT_HEADLESS: yes
+          PLAYWRIGHT_BROWSER: ${{ matrix.browser }}
+          PLAYWRIGHT_TRACE_PATH: ${{ matrix.browser }}-playwright-trace.zip
+          SERVE_FRONTEND: yes 
```

Comment:
Maybe also explicitly set E2E_PORT? :thinking: Maybe we should also get the frontend-build use this setting :thinking: